nixos/modules/services/hardware/fancontrol.nix

```diff
@@ -0,0 +1,39 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.fancontrol;
+
+in {
+
+  options.hardware.fancontrol = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Whether to enable fancontrol (requires a configuration file, see pwmconfig)";
+    };
+
+    configFile = mkOption {
```

Comment:
This leads to an impure NixOS configuration. The config file should reside in the nix store.

Reply:
the config file is generated by the user with pwmconfig, which by default selects /etc/fancontrol
i don't know how to clean that up...

Reply:
You could change the type to `lines` and then use `pgks.writeText` to write the content of `configFile` (maybe rename it to config?) to a file in the nix store. The user can still use `pwmconfig` to generate the config but then puts its output into their `configuration.nix`.

Reply:
if a fallback to /etc/fancontrol can be used, it could still "just work"
is there any documentation on the pattern you're suggesting, or do you know of an exemplary module?
configFile seems to be the conventional name for configuration files

Reply:
> if a fallback to /etc/fancontrol can be used, it could still "just work"

That would be against NixOS design principles. All config files in /etc should be maintained by NixOS. Allowing for manually maintained fallback files will lead to non-reproducible system.

For examples see `nixos/modules/tasks/filesystems/nfs.nix`  or `nixos/modules/services/amqp/rabbitmq.nix`

Reply:
`nixos/modules/services/web-servers/mighttpd2.nix` proved more instructive